cdap-master/src/main/java/io/cdap/cdap/master/environment/k8s/MetadataServiceMain.java

```diff
@@ -98,6 +100,7 @@ protected void configure() {
           bind(OwnerAdmin.class).to(DefaultOwnerAdmin.class);
           // TODO (CDAP-14677): find a better way to inject metadata publisher
           bind(MetadataPublisher.class).to(MessagingMetadataPublisher.class);
+          bind(MetadataServiceClient.class).to(DefaultMetadataServiceClient.class);
```

Comment:
Does MetadataService itself depend on the metadata service client? Does it make HTTP calls to itself? I hope not, and then we should not need this?